### PR TITLE
Introduced `True`, `IsTrue`, `IsFalse`, `IsMember` and `IsNotMember` specifications

### PR DESCRIPTION
**Pull request by gotfrydoh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -235,6 +235,60 @@ to return customers without orders (with empty collection of orders).
...
+
+A negation for this specification is also available: `IsFalse`.
+
+### IsMember ###
+
+Filters with particular member in collection defined under `path` in `@Spec` annotation.
+
+Supports collections of multiple data types: numbers, booleans, strings, dates, enums.
+
+For example, for `luckyNumbers` table which contains association to `customers` table (one customer can have multiple lucky numbers) the following mapping can be introduced:
```
Comment (tkaczmarzyk) [inline README.md:280]: Let's simplify it:  For example, let's assume a Customer entity with lukcyNumber collection attribute (one customer can have multiple lucky numbers). The following mapping can be introduced:

File: README.md [hunk 2]
```diff
@@ -235,6 +235,60 @@ to return customers without orders (with empty collection of orders).
...
+
+to return all special customers (with `gold` field set to `true`).
+
+A negation for this specification is also available: `IsFalse`.
+
+### IsMember ###
+
+Filters with particular member in collection defined under `path` in `@Spec` annotation.
+
+Supports collections of multiple data types: numbers, booleans, strings, dates, enums.
```
Comment (tkaczmarzyk) [inline README.md:278]: I'm assuming that this works only for @ELementCollection, not @oneTomany ? If so, then it's worth mentioning it.

File: src/main/java/net/kaczmarzyk/spring/data/jpa/domain/IsFalse.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+import net.kaczmarzyk.spring.data.jpa.utils.QueryContext;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+/**
+ * <p>Filters using `false` value for a boolean type field, i.e. tests for a false value.</p>
```
Comment (tkaczmarzyk) [inline src/main/java/net/kaczmarzyk/spring/data/jpa/domain/IsFalse.java:26]: please also add the reference to Equal (see my previous comments, to be coped from readme)

File: src/main/java/net/kaczmarzyk/spring/data/jpa/domain/IsMember.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Objects;
+
+/**
+ * <p>Filters with particular member in collection defined under `path` in `@Spec` annotation.</p>
+ *
+ * @author Hubert Gotfryd (Tratif sp. z o.o.)
+ */
+public class IsMember<T> extends PathSpecification<T> {
+
+    private String expectedValue;
```
Comment (tkaczmarzyk) [inline src/main/java/net/kaczmarzyk/spring/data/jpa/domain/IsMember.java:36]: expectedMember?

**Review by tkaczmarzyk — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 5]
```diff
@@ -231,6 +231,10 @@
 			<groupId>javax.persistence</groupId>
 			<artifactId>javax.persistence-api</artifactId>
 		</dependency>
+		<dependency>
+			<groupId>org.hibernate</groupId>
+			<artifactId>hibernate-core</artifactId>
+		</dependency>
```
[the remark this answers is not in the document]
Comment (gotfrydoh) [reply pom.xml:237]: We need `PluralAttributePath` from hibernate to obtain information about data type of element in Collection for `isMember` and `isNotMember` specifications. 
`Class<?> typeOnPath = ((PluralAttributePath<Object>) path(root)).getAttribute().getElementType().getJavaType();`

**Review by tkaczmarzyk — APPROVED** [review]

**Review by tkaczmarzyk — APPROVED** [review]

**Pull request merged**